pages/index.js

```diff
@@ -112,7 +112,7 @@ const argTypes = {
 export default function Home() {
   const [args, { updateAt: updateArg, removeAt: removeArg, insertAt: insertArg }] = useList([
     { type: argTypes.MODEL, value: "ExampleModel" },
-    { type: argTypes.ATTRIBUTE, value: "other_model:references{polymorphic}:uniq" },
+    { type: argTypes.ATTRIBUTE, value: "other_model:references{polymorphic}:uniq", valid: true},
```

Comment:
This is a good initiative!

I don't quite feel comfortable that `valid` becomes a prop, though it means the parent component can decide whether a value is valid or not. Maybe we can push that responsibility to the child? So the `FieldInput` component (which needs a rename) can maintain its own state of the input, validate the input as a user types, and only trigger `onUpdate` when it's valid. It does mean that the parent won't be able to tell whether the value is valid or not, but maybe that's ok.

I'd suggest to remove this from this PR, we can address it more deeply in another PR. One library that I've heard of that helps handle validations is https://formik.org/, which is nice because we can validate the inputs. It seems like that requires the user to trigger a full form submission though, whereas I kinda like the current "enter the value and see it updated in the command immediately" experience. We might be able to get some inspiration from there though.

Reply:
True, didn't think of it that way. Thanks! will update this.